### PR TITLE
Make it easier to try out the demo

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,240 @@
...
+How it is going to be triggered (system property, command-line argument etc.) is not decided yet.
+
+You can see more bytecode upgrades happening now:
+
+```text
+Matched call to org.gradle.language.jvm.tasks.ProcessResources.setDestinationDir(java.io.File) in test.project.build.TestGroovyPlugin$_apply_closure1, replacing...
+Matched call to org.gradle.api.tasks.compile.CompileOptions.getCompilerArgs() in test.project.build.TestGroovyPlugin$_apply_closure2, replacing...
+```
+
+These are running against the bytcode of the build logic that wes compiled against Gradle 7.4.1, making the resulting bytecode comaptible with Gradle Next.
```
Comment (wolfs) [inline README.md]:
```suggestion
These are running against the bytecode of the build logic that wes compiled against Gradle 7.4.1, making the resulting bytecode compatible with Gradle Next.
```

typo

File: test-project.patch [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+<+>UTF-8
+===================================================================
+diff --git a/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle b/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle
+--- a/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle	(revision c88205e9ca573c625403202c42a9e8f64656e431)
++++ b/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle	(date 1652965995179)
+@@ -8,7 +8,7 @@
+ tasks.withType(AbstractCompile).configureEach {
+     targetCompatibility = "11"
+     sourceCompatibility = "11"
+-    classpath += files()
```
Comment (wolfs) [inline test-project.patch]: Is this change necessary? I don't remember if I fixed this or not.

File: test-project.patch [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+===================================================================
+diff --git a/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt b/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt
+--- a/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt	(revision c88205e9ca573c625403202c42a9e8f64656e431)
++++ b/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt	(date 1652965995189)
+@@ -9,7 +9,7 @@
+ class TestKotlinPlugin : Plugin<Project> {
+     override fun apply(project: Project) {
+         project.tasks.withType<ProcessResources>().configureEach {
+-            destinationDir = project.layout.buildDirectory.dir("new-resources").get().asFile
++            destinationDir = project.layout.buildDirectory.dir("new-resources")
```
Comment (wolfs) [inline test-project.patch:91]: This change isn't really necessary for the upgrade, though it improves things, right?

**Review by wolfs — APPROVED** [review]
Looks good, I lefts some questions.

File: test-project/build.gradle.kts [hunk 4]
```diff
@@ -1,15 +0,0 @@
-
-abstract class MyTask : DefaultTask() {
```
Comment (wolfs) [inline test-project/build.gradle.kts:2]: This file is important since it is used to show how the IDE support works.

File: test-project.patch (hunk 2 above)
Comment (lptr) [reply test-project.patch]: I checked and the build works without making this change, too. Fixed.

File: test-project.patch (hunk 3 above)
Comment (lptr) [reply test-project.patch:91]: I added comments to explain each change. I wonder if we should add comments in places where we did _not_ have to change things..?

File: test-project/build.gradle.kts (hunk 4 above)
Comment (lptr) [reply test-project/build.gradle.kts:2]: I think that part of the demo should work with the other build logic files. Can we move this logic to one of them?

Comment (wolfs) [reply test-project/build.gradle.kts:2]: We can move it to the other build logic files, though it seems to be more distracting there. And if you don't want to look into these changes here, then you can ignore this file completely. See the video I recorded what we show with these changes.

File: test-project.patch (hunk 3 above)
Comment (wolfs) [reply test-project.patch:91]: I think the comments in the places where we didn't need to change things would be good as well.

**Review by asodja — APPROVED** [review]

File: test-project/build.gradle.kts (hunk 4 above)
Comment (lptr) [reply test-project/build.gradle.kts:2]: My only problem is that it's in the root of `test-project` and when I was first looking at the repo, it got me pretty confused as to why it's here. If we could move it to some other directory, that'd already be an improvement.

Comment (lptr) [reply test-project/build.gradle.kts:2]: If we could make this file part of the build somehow, then the video could be made better, too, I think. Now there is a big break between the first part and the second part for me.

Comment (wolfs) [reply test-project/build.gradle.kts:2]: Thinking about this again, we can also remove it. This was about showing the IDE support for the assign operator better, though we'll get better IDE support and then we can show it directly in the already existing Kotlin build scripts.

File: test-project.patch (hunk 3 above)
Comment (lptr) [reply test-project.patch:91]: Added.

File: test-project.patch [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+--- a/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle
++++ b/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle
+@@ -6,8 +6,10 @@ plugins {
+ pluginManager.apply(test.project.build.TestGroovyPlugin)
+ 
+ tasks.withType(AbstractCompile).configureEach {
++    // Keeps working because dynamic Groovy handles `Property` assignment
+     targetCompatibility = "11"
+     sourceCompatibility = "11"
++    // Keeps workign via `FileCollection.plus()`
```
Comment (wolfs) [inline test-project.patch]:
```suggestion
+    // Keeps working via `FileCollection.plus()`
```

**Review by wolfs — APPROVED** [review]
